lib/chef/knife/azurerm_base.rb

```diff
@@ -98,7 +102,31 @@ def authentication_details
         token_details
       end
 
+      def current_xplat_cli_version
+        Mixlib::ShellOut.new("azure -v").run_command.stdout
```

Comment:
I see this is the common case in knife-azure, but we never run `Mixlib::ShellOut.error!` so we're going to silently continue here if 'azure' doesn't exist and return an empty string, and then `is_old_xplat?` is going to return true when comparing an empty string with `XPLAT_VERSION_WITH_WCM_DEPRECATED`.

I wonder if requiring `chef/mixin/shell_out` and then using `shell_out!` would be an easier pattern to establish than checking `error!`. Either way, we should validate we got reasonable output here in a way that gives some debugging information if we don't.